Tools/JFusion/Program.cs

```diff
@@ -1,88 +1,152 @@
+                            mfaFProj.Write($"output\\{JOutN}");
+                            Console.WriteLine("Json writing finished");
+                        }
+                        catch when (!File.Exists(mfapath))
+                        {
+
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine("ERROR: File not found");
+                            // goto MFAPATH;
+                        }
+                    }
+                    break;
+
+
+                case ConsoleKey.D2:
+                    while (true)
+                    {
+                        try
+                        {
+                            Console.ForegroundColor = ConsoleColor.Green;
+                            Console.Write("Json path: ");
+                            jpath = Console.ReadLine().Trim('"');
+                            Console.ForegroundColor = ConsoleColor.Green;
+                            string MOutN = string.Empty;
+                            Console.WriteLine("Output Name: ");
+                            MOutN = Console.ReadLine();
+                            Console.ForegroundColor = ConsoleColor.Green;
+                            Console.WriteLine("Converting Json to MFA");
+                            Console.WriteLine("Reading Json...");
+                            var ogFProj = JMFAFile.Open(jpath);
+                            Console.WriteLine("Json reading finished.");
+                            Console.WriteLine("Converting Json...");
+                            var mfa = ogFProj.ToMFA();
+                            Console.WriteLine("Json conversion finished.");
+                            Console.WriteLine("Writing MFA...");
+                            mfa.Write(new ByteWriter($"output\\{MOutN}.mfa", FileMode.Create));
+                            Console.WriteLine("MFA writing finished");
+                        }
+                        catch when (!File.Exists(mfapath))
+                        {
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine("ERROR: Project not found");
+                        }
+                    }
+                    break;
+
+
+                case ConsoleKey.D3:
+                    var testMfa = new MFAData();
+                    testMfa.Read(new ByteReader("test.mfa", FileMode.Open));
+                    testMfa.Write(new ByteWriter("test_out.mfa", FileMode.Create));
+                    break;
+                default:
+                    goto SELECT_MODE;
+            }
+        }
+        catch (Exception ex)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine(ex);
+            while (true)
+            {
+                if (Console.ReadKey().Key == ConsoleKey.Enter) break;
+            }
+        }
+        goto SELECT_MODE;
+

```

Comment:
In this case, you can get rid of while loops completely, code loops itself by it's on